### ISSUE TITLE
Easy way to expand upon the existing webpack.config.js?

### ISSUE THREAD

**Issue by stephan-v** [JeffreyWay/laravel-mix#634]
I am working on a non-Laravel project where I have integrated laravel-mix all is fine and well until I have to update. Lately a lot of cool and interesting things have been added to laravel-mix but each and everytime I have to copy over the webpack.config.js from the laravel-mix node_modules directory to get all the new additions.

```
cp -r node_modules/laravel-mix/setup/** ./
```

https://github.com/JeffreyWay/laravel-mix/blob/master/docs/installation.md#stand-alone-project

This would normally not be an issue when I would have a clean webpack.config.js file, but I have edited it to include ESLint which forces me to first copy over the file from the node_modules directory and in turn copy over the bits and pieces I had added manually to my old webpack.config.js file.

Is there an easier way to deal with this workflow? Maybe create another webpack file that expands upon the config file from laravel-mix or something? Hopefully someone can supply a bit of input on this.

Cheers.

**Comment by QWp6t** [comment]
https://github.com/JeffreyWay/laravel-mix/blob/master/docs/quick-webpack-configuration.md

**Comment by stephan-v** [comment]
This works for laravel projects since the laravel-mix webpack file is probably known within Laravel, I doubt this works for non-laravel projects though. How would my app even know where to find that file from within my extended webpack file?

I doubt you can simply deep merge like provided in the documentation with a non-laravel application. Correct me if I am wrong on this though. I will try it out in a sec.

**Comment by stephan-v** [comment]
Got it working:

```
mix.webpackConfig({
    devServer: {
        overlay: true,
    },
    module: {
        rules: [
            {
                test: /.(vue|js)$/,
                loader: 'eslint-loader',
                enforce: 'pre',
                exclude: /node_modules/
            }
        ]
    }
});
```

The major key is that my app needed to know where to find the original webpack config file from the `laravel-mix` node_modulse directory. I found the solution for this inside another issue:

I updated my `npm run watch` to this:

`"watch": "cross-env NODE_ENV=development webpack --watch --progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js",`

the `--config` option provided a way to point to the webpack.config.js file I want to extend. Possibly this could be a good addition to the documentation since there is no way to easily know this?

It seems by default the laravel package.json has a pretty good example for the setup which I do not initially see. Possibly it could be handy to point people manually installing to this file as a basic setup?

https://github.com/laravel/laravel/blob/master/package.json

Thanks for reading, cheers.
